### ISSUE TITLE
QuadtreeRasterOverlay sub-tile cache size accounting is wrong

### ISSUE THREAD

**Issue by kring** [CesiumGS/cesium-native#958]
`QuadtreeRasterOverlay` has a small cache that is intended to briefly store decoded raster overlay tile images that span multiple geometry tiles, so we don't need to decode them multiple times (or download them multiple times in the hopefully-unlikely event that the requests aren't cached).

But it doesn't appear to be working correctly.

`QuadtreeRasterOverlayTileProvider` has a `_cachedBytes` field. When a new overlay image is loaded, this field is immediately incremented accordingly. The code that later _decrements_ this field looks like this:

```cpp
this->_tileLookup.erase(it->tileID);
it = this->_tilesOldToRecent.erase(it);

// If this is the last use of this data, it will be freed when the shared
// pointer goes out of scope, so reduce the cachedBytes accordingly.
if (pImage.use_count() == 1) {
  if (pImage->image) {
    this->_cachedBytes -= int64_t(pImage->image->pixelData.size());
    CESIUM_ASSERT(this->_cachedBytes >= 0);
  }
}
```

This snippet is executed when we're removing this tile from the cache because the cache is full and this tile is the one that is least-recently-used. It is _not_ necessarily true that this image is unused when this happens. And because of that `use_count() == 1` check, if this tile happens to be in use elsewhere when it's evicted from the cache, we'll never subtract its bytes.

As a result, over time (and it doesn't take very long), we'll think the cache is completely full at all times and each new image will be evicted immediately, making the cache pretty close to useless.

I think the solution is simply to remove that `use_count() == 1` check. When we evict an image from the cache, we should always reduce the number of bytes in the cache accordingly. Based on the comment, when I wrote this, I seemed to be hung up on the question of whether the image would actually be freed. But the freeing doesn't matter; all that matters is whether this image is kept in the set of tiles temporarily kept around even though they're potentially unused.

**Comment by kring** [comment]
This entire sub-tile cache should probably just be replaced by the external image cache in #926!

**Comment by kring** [comment]
Fixed in #992.
